src/gov/nist/javax/sip/stack/SIPClientTransactionImpl.java

```diff
@@ -1545,7 +1544,7 @@ public void processResponse(SIPResponse sipResponse, MessageChannel incomingChan
     int code = sipResponse.getStatusCode();
     boolean isRetransmission = !responsesReceived.add(Integer.valueOf(code));
     if (code > 100 && code < 200 && isRetransmission) {
-      if (lastResponse != null && !sipResponse.toString().equals(lastResponse.toString())) {
+      if (lastResponse != null && !sipResponse.equals(lastResponse)) {
```

Comment:
Equals will compare the statusLine and the headers, not the body. In original issue is reported that "not deliver events to JSLEE application when provisional response (100 -200) is repeated with new content or header". Is this patch taking into account those "header" modification details...? I just see this fixing the "content" modifications...

Do you have a unit test to prove this working?